### PR TITLE
Merge params/* and remove index files and re-exports

### PR DESCRIPTION
**Pull request by kainino0x**
This:
- Makes it so there's only one true way to import anything
- Prevents IDE from assuming it's Node and generating an import of the
  directory containing an `index.ts` (e.g. 'path/to/' for 'path/to/index.js').
- Without blowing up the total number of imports too much

### CODE REVIEW

File: src/webgpu/api/validation/setBindGroup.spec.ts [hunk 1]
```diff
@@ -2,7 +2,9 @@ export const description = `
 setBindGroup validation tests.
 `;
 
-import { TestGroup, pcombine, poptions } from '../../../common/framework/index.js';
+import { pcombine } from '../../../common/framework/params.js';
```
Comment (austinEng) [inline src/webgpu/api/validation/setBindGroup.spec.ts:5]: nit: merge these

File: src/unittests/param_helpers.spec.ts [hunk 2]
```diff
@@ -2,16 +2,11 @@ export const description = `
...
-  TestGroup,
-  paramsEquals,
-  pcombine,
-  pexclude,
-  pfilter,
-  poptions,
-} from '../common/framework/index.js';
+import { pcombine, poptions } from '../common/framework/params.js';
+import { pexclude } from '../common/framework/params.js';
+import { pfilter } from '../common/framework/params.js';
```
Comment (austinEng) [inline src/unittests/param_helpers.spec.ts:7]: nit: merge these

File: src/unittests/params.spec.ts [hunk 3]
```diff
@@ -2,7 +2,11 @@ export const description = `
 Unit tests for parameterization.
 `;
 
-import { ParamSpec, TestGroup, pcombine, pexclude, pfilter } from '../common/framework/index.js';
+import { pcombine } from '../common/framework/params.js';
+import { pexclude } from '../common/framework/params.js';
+import { pfilter } from '../common/framework/params.js';
```
Comment (austinEng) [inline src/unittests/params.spec.ts:7]: merge

File: src/webgpu/api/operation/buffers/create_mapped.spec.ts [hunk 4]
```diff
@@ -1,6 +1,8 @@
 export const description = ``;
 
-import { TestGroup, pbool, pcombine, poptions } from '../../../../common/framework/index.js';
+import { pcombine } from '../../../../common/framework/params.js';
+import { pbool, poptions } from '../../../../common/framework/params.js';
```
Comment (austinEng) [inline src/webgpu/api/operation/buffers/create_mapped.spec.ts:4]: merge

File: src/webgpu/api/operation/buffers/map.spec.ts [hunk 5]
```diff
@@ -1,6 +1,8 @@
 export const description = ``;
 
-import { TestGroup, pbool, pcombine, poptions } from '../../../../common/framework/index.js';
+import { pcombine } from '../../../../common/framework/params.js';
+import { pbool, poptions } from '../../../../common/framework/params.js';
```
Comment (austinEng) [inline src/webgpu/api/operation/buffers/map.spec.ts:4]: merge

File: src/webgpu/api/validation/createBindGroup.spec.ts [hunk 6]
```diff
@@ -2,7 +2,11 @@ export const description = `
 createBindGroup validation tests.
 `;
 
-import { C, TestGroup, pcombine, poptions, unreachable } from '../../../common/framework/index.js';
+import * as C from '../../../common/constants.js';
+import { pcombine } from '../../../common/framework/params.js';
+import { poptions } from '../../../common/framework/params.js';
```
Comment (austinEng) [inline src/webgpu/api/validation/createBindGroup.spec.ts:7]: merge

File: src/webgpu/api/validation/createPipelineLayout.spec.ts [hunk 7]
```diff
@@ -2,7 +2,9 @@ export const description = `
 createPipelineLayout validation tests.
 `;
 
-import { TestGroup, pbool, pcombine, poptions } from '../../../common/framework/index.js';
+import { pcombine } from '../../../common/framework/params.js';
+import { pbool, poptions } from '../../../common/framework/params.js';
```
Comment (austinEng) [inline src/webgpu/api/validation/createPipelineLayout.spec.ts:6]: merge

File: src/webgpu/web-platform/copyImageBitmapToTexture.spec.ts [hunk 8]
```diff
@@ -2,7 +2,9 @@ export const description = `
 copy imageBitmap To texture tests.
 `;
 
-import { TestGroup, pcombine, poptions } from '../../common/framework/index.js';
+import { pcombine } from '../../common/framework/params.js';
+import { poptions } from '../../common/framework/params.js';
```
Comment (austinEng) [inline src/webgpu/web-platform/copyImageBitmapToTexture.spec.ts:6]: merge

**Comment by kainino0x** [comment]
oops, knew I needed to check those but was hoping the linters would fix them for me.

**Pull request merged**